Tutorial: How to set up A/B/n testing (#6543)
PostHog/posthog.com: contents/tutorials/abn-testing.md

In our a `page.js` file, we: 

1. Import `useFeatureFlagVariantKey` from `posthog-js/react`, and `useState` and `useEffect` from `react`
2. Set up a state for `mainCopy` using `useState`. 
3. Set the`mainCopy` state in a `useEffect` using the variant key value we get from PostHog.
4. Show the `mainCopy` state to users in our component.

Together, this looks like this:

```js
// app/page.js
'use client'
import { useFeatureFlagVariantKey } from 'posthog-js/react'
import { useState, useEffect } from 'react'

export default function Home() {

  const [mainCopy, setMainCopy] = useState('Welcome to our app')
  const copyVariant = useFeatureFlagVariantKey('homepage-abn')
  
  useEffect(() => {
    if (copyVariant && copyVariant !== 'control') {
      setMainCopy(`Welcome to our ${copyVariant} app`);
    }
  }, [copyVariant])

  return (
    <div>
      <h1>{mainCopy}</h1>
      <p>This is a page being A/B/n tested</p>
    </div>
  )
}
```

When we reload our page, we get a title based on the flag key variant. 

![Epic webpage](../images/tutorials/abn-testing/epic.png)

> ### Checking the other variants

Reviewer: ```suggestion
### Testing the other variants
```

> We can check the other variants by going to the [feature flag page](https://app.posthog.com/feature_flags), searching for the key related to our experiment, and clicking on it to edit it. Scroll down to release conditions and set the **optional override** to any of the flag values. If you want the optional override to apply to only you, you can create another condition set. Finally, press save and when you go back to your app, you see the overridden flag value.

Reviewer: Need a bit more details on how to make it only apply to you. For this tutorial, since we're not using identify, maybe something like utm_source is better

```suggestion
We can check the other variants by going to the [feature flag page](https://app.posthog.com/feature_flags), searching for the key related to our experiment, and clicking on it to edit it. Scroll down to release conditions and set the **optional override** to any of the flag values. 

If you want the optional override to apply to only you, you can create another condition set. Finally, press save and when you go back to your app, you see the overridden flag value.
```

![Overriding our variant](../images/tutorials/abn-testing/override.png)

> **Note:** Make sure to remove the optional override before rolling out your A/B/n test to real users.

### Rolling out the winning variant

> Once your experiment reaches significance, you can stop it and roll out the winning variant. 

Reviewer: I wouldn't say significance, since this is a new concept that we havent introduced yet in this tuotiral.
```suggestion
Once your experiment has concluded, you can stop it and roll out the winning variant. 
```
